### ISSUE TITLE
Crash if system's default SMS/MMS app has been disabled

### ISSUE THREAD

**Issue by acidstout** [WhisperSystems/Signal-Android#5828]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
Signal crashes when trying to compose a new message or to read a received message, if the system's default SMS/MMS app is disabled.

### Steps to reproduce
- Have Signal installed
- Disable the system's default SMS/MMS app in the system Settings -> Apps -> default Android SMS/MMS app -> Disable / Deactivate (I use Android in german language, so actually I can only guess what's it called on Android in english).
- Open Signal and try to compose a new message by clicking on a contact or on the compose button in the lower right corner and then select a contact from the list.

**Actual result:**
Signal stops responding for appr. 2-3 seconds and then a force close is triggered and Android asks me if I want to close the process or send a bug report.
**Expected result:**
If I click the compose button and then select a contact to send a message to, or if I directly click on a contact or and existing conversation, Signal should open the conversation view and allow me to see the 
existing conversation (if applicable) and allow me to type in my message.

### Device info
**Device:** Motorola Moto E (2014 / XT1021)
**Android version:** 4.4.2 (Retail Stock ROM)
**Signal version:** 3.21.0

### Link to debug log
https://gist.github.com/anonymous/cacedef0ad54400a8038ff08d3e91199

**Comment by FeuRenard** [comment]
Did you set Signal as new default SMS/MMS app after you disabled the system's SMS/MMS app?
In Signal: Settings -> SMS + MMS -> SMS enabled/disabled

**Comment by McLoo** [comment]
Don't disable it.
https://github.com/WhisperSystems/Signal-Android/issues/1110

**Comment by acidstout** [comment]
Sorry for my late response. Last months have been very busy.

So, yes, Signal is my default SMS/MMS app. It always was from the time when I first installed Signal, which was named TextSecure then.

After reading #1110 I understand that it needs to be enabled in order for Signal to be able to receive/send unecrypted SMS/MMS, if I got it right. Also on Android 6.0.1 I just noticed that the system's SMS/MMS app cannot be disabled anymore.
